ipn/ipnlocal/local.go

```diff
@@ -1170,6 +1170,15 @@ func (b *LocalBackend) SetControlClientStatus(c controlclient.Client, st control
 
 	unlock.UnlockEarly()
 
+	var shouldSuggest bool
+	if st.NetMap != nil {
+		for _, peer := range st.NetMap.Peers {
+			if shouldAutoExitNode() && peer.StableID() == b.Prefs().ExitNodeID() && peer.Online() != nil && !*peer.Online() {
+				shouldSuggest = true
+			}
+		}
+	}
```

Comment:
I think that's along the right lines, but I think it'll work better slightly differently

1. Move `shouldAutoExitNode()` out to the `st.NetMap != nil` condition, so no loop happens if auto exit node isn't in use. Perhaps also check for `b.lastSuggestedExitNode != ""`.
2. Compare `peer.StableID()` against `b.lastSuggestedExitNode`. You'll need to move this before the unlock for that to be safe.
3. Rather than setting `shouldSuggest`, I think it is sufficient to set `b.lastSuggestedExitNode = ""` and then check for that later.
4. Break the loop once you find a match; there won't be any other matches.

Side note: `b.Prefs()` is expensive (it locks, gets the prefs, copies them and removes sensitive data, then unlocks and returns), so it should be done the minimum number of times required (i.e. save it locally before starting the loop). Ideally, if working with prefs, that should happen a few lines down where it grabs a copy of the full prefs and makes changes in bulk.